packages/nvidia_nat_data_flywheel/src/nat/plugins/data_flywheel/observability/processor/trace_conversion/span_to_dfw_record.py

```diff
@@ -0,0 +1,118 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from enum import Enum
+from typing import Any
+
+from pydantic import BaseModel
+
+from nat.data_models.span import Span
+from nat.plugins.data_flywheel.observability.processor.trace_conversion.trace_adapter_registry import (
+    TraceAdapterRegistry,  # noqa: F401
+)
+from nat.plugins.data_flywheel.observability.schema.trace_container import TraceContainer
+
+logger = logging.getLogger(__name__)
+
+
+def _get_string_value(value: Any) -> str:
+    """Extract string value from enum or literal type safely.
+
+    Args:
+        value (Any): Could be an Enum, string, or other type
+
+    Returns:
+        str: String representation of the value
+    """
+    if isinstance(value, Enum):
+        return value.value
```

Comment:
If an enum is non-str based, then you need to wrap
```suggestion
        return str(value.value)
```